### PR TITLE
PTV-1733: Consolidate* reference parsing code

### PR DESCRIPTION
**Pull request by MrCreosote**
Turns out I'll need a reference path validator for the provenance work.
The first step is to make a single parser rather than having it split
into two chunks. Improves code cohesion as well.

\* mostly, there's still some stuff in IdentifierUtils but it does weird
things around rearranging the ref order that I don't want to touch for
now

### CODE REVIEW

File: src/us/kbase/workspace/database/Workspace.java [hunk 1]
```diff
@@ -1803,34 +1803,14 @@ private ObjectIdentifier parseIDString(
...
+			try {
+				return ObjectIdentifier.getBuilderFromRefPath(id).build();
+				//Illegal arg is probably not the right exception
+			} catch (IllegalArgumentException e) {
+				final List<String> attribs = getAnyAttributeSet(associatedObject, id);
+				throw new IdParseException(
+						e.getMessage(), getIdType(), associatedObject, id, attribs, e);
 			}
-			// probably not worth making a builder starting from a reference to avoid an
-			// extra OI instantiation
```
Comment (MrCreosote) [inline src/us/kbase/workspace/database/Workspace.java:1830]: Well this aged like milk

File: src/us/kbase/workspace/database/provenance/Common.java [hunk 2]
```diff
@@ -10,6 +10,8 @@
  * Tests are all via testing the public provenance class APIs.
  */
 class Common {
+	
+	private Common() {};
```
Comment (MrCreosote) [inline src/us/kbase/workspace/database/provenance/Common.java:14]: Should get codecov to shut up about the uninstantiated class maybe?

File: src/us/kbase/workspace/kbase/WorkspaceServerMethods.java [hunk 3]
```diff
@@ -442,8 +442,6 @@ public GetObjects2Results getObjects(
 					InaccessibleObjectException, NoSuchReferenceException, NoSuchObjectException,
 					TypedObjectExtractionException, ReferenceSearchMaximumSizeExceededException,
 					InterruptedException {
-		// ugh, tried to see if we could code out the jpe and ioe but it's too much of a mess
-		// once you get into the byte array cache and JTS
```
Comment (MrCreosote) [inline src/us/kbase/workspace/kbase/WorkspaceServerMethods.java:446]: Well, they're gone now, so I guess I unmessed it at some point

File: src/us/kbase/workspace/test/workspace/ReferenceTest.java [hunk 4]
```diff
@@ -32,13 +32,14 @@ public void mongoRef() throws Exception {
...
-		failMakeRef("1", "Illegal number of separators / in object reference 1");
+		failMakeRef("1", "Illegal number of separators '/' in object reference '1'");
 		failMakeRef("1/2", "ref 1/2 is not an absolute reference");
 		failMakeRef("1/foo", "ref 1/foo is not an absolute reference");
 		failMakeRef("foo/1/2", "ref foo/1/2 is not an absolute reference");
 		failMakeRef("1/foo/2", "ref 1/foo/2 is not an absolute reference");
-		failMakeRef("1/1/foo", "Unable to parse version portion of object reference 1/1/foo to an integer");
-		failMakeRef("1/2/3/4", "Illegal number of separators / in object reference 1/2/3/4");
+		failMakeRef("1/1/foo",
+				"Unable to parse version portion of object reference 1/1/foo to an integer");
```
Comment (ialarmedalien) [inline src/us/kbase/workspace/test/workspace/ReferenceTest.java:41]: no `'` around the object reference here?

File: src/us/kbase/workspace/test/workspace/WorkspaceTest.java [hunk 5]
```diff
@@ -3207,7 +3207,7 @@ public void wsIDHandling() throws Exception {
 		innertuple.set(0, Arrays.asList("foo", "YourMotherWasAHamster"));
 		failSave(user, wsi, objs, new TypedObjectValidationException(
 				"Object #1, foo has unparseable reference YourMotherWasAHamster: Illegal number " +
-				"of separators / in object reference YourMotherWasAHamster at /ws_2/0/1"));
+				"of separators '/' in object reference 'YourMotherWasAHamster' at /ws_2/0/1"));
```
Comment (ialarmedalien) [inline src/us/kbase/workspace/test/workspace/WorkspaceTest.java:3210]: 🐹

**Review by ialarmedalien — APPROVED** [review]
Looks good!

File: src/us/kbase/workspace/test/workspace/ReferenceTest.java (hunk 4 above)
Comment (MrCreosote) [reply src/us/kbase/workspace/test/workspace/ReferenceTest.java]: Done. Didn't actually touch the code that throws this error, but yeah, should be consistent

**Pull request merged**